src/macaron/slsa_analyzer/checks/provenance_verified_check.py

```diff
@@ -0,0 +1,118 @@
+# Copyright (c) 2024 - 2024, Oracle and/or its affiliates. All rights reserved.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl/.
+
+"""This module adds a Check that checks whether the provenance is verified."""
+import logging
+
+from sqlalchemy import ForeignKey, String
+from sqlalchemy.orm import Mapped, mapped_column
+
+from macaron.database.table_definitions import CheckFacts
+from macaron.json_tools import json_extract
+from macaron.slsa_analyzer.analyze_context import AnalyzeContext
+from macaron.slsa_analyzer.checks.base_check import BaseCheck
+from macaron.slsa_analyzer.checks.check_result import CheckResultData, CheckResultType, Confidence, JustificationType
+from macaron.slsa_analyzer.registry import registry
+from macaron.slsa_analyzer.slsa_req import ReqName
+
+logger: logging.Logger = logging.getLogger(__name__)
+
+
+class ProvenanceVerifiedFacts(CheckFacts):
+    """The ORM mapping for justifications in the provenance verified check."""
+
+    __tablename__ = "_provenance_verified_check"
+
+    #: The primary key.
+    id: Mapped[int] = mapped_column(ForeignKey("_check_facts.id"), primary_key=True)  # noqa: A003
+
+    #: The SLSA build level of the provenance.
+    build_level: Mapped[str] = mapped_column(String, nullable=True, info={"justification": JustificationType.TEXT})
```

Comment:
I think this should be an integer to allow checking the value of the level in policies. Also, any particular reason to set `nullable` to `True`?